compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1580,11 +1580,16 @@ impl<'tcx> ExistentialProjection<'tcx> {
     ) -> Self {
         // Assert there is a Self.
         projection_predicate.projection_ty.substs.type_at(0);
+        let ty = if let Term::Ty(ty) = projection_predicate.term {
+            ty
+        } else {
+            panic!("Only types are permitted here");
```

Comment:
what's the rationale for this? why can't there be constants here?

Reply:
Oops, I have no idea why I wrote this. I think it might've been because the existential predicate only accepts a type, so for now I only wanted to leave it for that case.